include/mxnet/lib_api.h

```diff
@@ -426,21 +432,30 @@ typedef void (*sparse_malloc_t)(void*, int, int, int, void**, int64_t**, int64_t
 
 #if defined(__NVCC__)
   typedef cudaStream_t mx_stream_t;
+  typedef curandStatePhilox4_32_10_t* mx_gpu_rand_pt;
 #else
   typedef void* mx_stream_t;
+  typedef void* mx_gpu_rand_pt;
 #endif
 
+/*! \brief MXNet initialized random states for each device, used for parallelism */
+/* Each thread should generate random number unique sequence out of different states */
+#define NumCPURandomStates 1024
+#define NumGPURandomStates 32768
```

Comment:
Can we prefix these with MX_ and use all caps?
MX_CPU_RANDOM_STATES
MX_GPU_RANDOM_STATES